src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    case InvertIfStyle.MoveSubsequenceToElseBody:
+                        {
+                            var currentParent = ifNode.Parent;
+                            var statements = GetStatements(currentParent);
+                            var index = statements.IndexOf(ifNode);
+
+                            var statementsBeforeIf = statements.Take(index);
+                            var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                            var updatedIf = ifNode
+                                .WithCondition(negatedCondition)
+                                .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf));
+
+                            var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf))
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    case InvertIfStyle.WithElseClause:
+                        {
+                            var currentParent = ifNode.Parent;
+                            var statements = GetStatements(currentParent);
+                            var index = statements.IndexOf(ifNode);
+
+                            var statementsBeforeIf = statements.Take(index);
+                            var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                            var updatedIf = ifNode
+                                .WithCondition(negatedCondition)
+                                .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf))
+                                .WithElse(SyntaxFactory.ElseClause(ifNode.Statement));
+
+                            var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf))
+                                .WithAdditionalAnnotations(Formatter.Annotation);
+
+                            return root.ReplaceNode(currentParent, updatedParent);
+                        }
+
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(invertIfStyle);
+                }
+            }
+
+            private static IEnumerable<StatementSyntax> UnwrapBlock(StatementSyntax ifBody)
+            {
+                return ifBody is BlockSyntax block ? block.Statements : (IEnumerable<StatementSyntax>)new[] { ifBody };
+            }
+
+            private static StatementSyntax ReplaceEmbeddedStatement(StatementSyntax statement, StatementSyntax[] statements)
+            {
+                if (statements.Length > 0)
+                {
+                    statements[0] = statements[0].WithoutLeadingTrivia();
```

Comment:
these need explanations.  And they worry me in terms of losing comments users care about.

Reply:
Yup, couldn't find any way around it yet. trivia handling is *hard*. :(

Reply:
(pending)